### PR TITLE
Add "mechanism" in output.kafka to support  SCRAM-SHA-512 and  SCRAM-SHA-256 

### PR DESCRIPTION
**Pull request by zvictorino**
New config key `mechanism` was  introduced.
How to use it:
```
output.kafka:
  codec.format:
    string: '%{[@timestamp]} %{[message]}'

  hosts: ["localhost:9092"]
  topic: 'mytopic'
  partition.round_robin:
    reachable_only: false
  required_acks: 1
  compression: gzip
  max_message_bytes: 1000000
  username: user
  password: pass
  mechanism: SCRAM-SHA-512
```

### CODE REVIEW

**Comment by chlunde** [comment]
@jsoriano this would fix #8387

**Comment by jsoriano** [comment]
@zvictorino thanks for opening this PR! Tests are failing because there is a missing package, you will need to add it with `govendor fetch`.
It'd be also nice to add some docs for the new option, but we can add it in a follow up.

**Comment by urso** [comment]
let's name the setting `sasl_mechanism` or create an `sasl` namespace so the setting becomes `sasl.mechanism` (assuming we might add more settings in the future).

The setting will also need some docs.

Almost all symbols are private to the package, don't export them.

File: libbeat/outputs/kafka/config.go [hunk 1]
```diff
@@ -58,6 +89,7 @@ type kafkaConfig struct {
 	Username           string                    `config:"username"`
 	Password           string                    `config:"password"`
 	Codec              codec.Config              `config:"codec"`
+	Mechanism          string                    `config:"mechanism"`
```
Comment (urso) [inline libbeat/outputs/kafka/config.go]: this calls for an enum like type (over string):

```
type saslMechanism string

const (
  saslTypePlaintext = sarama.SASLTypePlaintext
  ...
)

func (m *saslMechanism) Unpack(in string) error {
  in = strings.ToUpper(in)  // try not to force users to use all upper case
  switch in {
  case saslTypePlaintext, ...:
    *m = saslMechanism(in)
  default:
    return fmt.Errorf("not valid mechanism '%v', only supported with PLAIN|SCRAM-SHA-512|SCRAM-SHA-256", in)
  }
  return nil   
}
```

The unpack method will be called by `(*Config).Unpack`. If it fails the full config name and the config file the setting comes from will be reported.

There is another 'string' comparison in order to fill in the sarama SASL settings. Having magic strings in different places is a good way to introduce Bugs (future developer might have typo when adding another mechanism). Better use an enum and use contanstants (compiler will complain if there is a typo). By having our own type we can also do this (optional):

```
func (m saslMechanism) configureSarama(... <other input>?, config *sarama.Config) error {
  switch m {
  case saslScramSHA256:
    ...

  case sasl<type>:
    ...

  default:
    panic(<we shouldn't get here, assuming no developer messed up>)
  }
}
```

**Comment by elasticcla** [comment]
Hi @zvictorino, we have found your signature in our records, but it seems like you have signed with a different e-mail than the one used in your Git [commit](https://github.com/elastic/beats/pull/12867.patch). Can you please add both of these e-mails into your Github profile (they can be hidden), so we can match your e-mails to your Github profile?

**Comment by ph** [comment]
Hello @zvictorino this is a really nice addition, its there anything we could help to move it forward with you?

**Comment by jsoriano** [comment]
jenkins, test this

**Comment by armysheng** [comment]
can anyone merge this pull request plz ?

**Comment by jsoriano** [comment]
@zvictorino there is a merge conflict, could you please update the branch with master? Thanks!

**Comment by jsoriano** [comment]
ok to test

**Comment by jsoriano** [comment]
jenkins, test this again please

**Comment by jsoriano** [comment]
Code LGTM, but there would be some pending things. We would need a changelog entry, docs for the new options, and some tests.

@zvictorino let me know if you can continue with this, if not I am ok with merging this as is and create an issue with some follow up tasks.

Thanks!

**Comment by jsoriano** [comment]
Follow up issue for this: https://github.com/elastic/beats/issues/16723

**Pull request merged**